src/backend/distributed/executor/citus_custom_scan.c

```diff
@@ -780,7 +780,13 @@ CitusEndScan(CustomScanState *node)
  */
 static void
 CitusReScan(CustomScanState *node)
-{ }
+{
+	CitusScanState *scanState = (CitusScanState *) node;
+	if (scanState->tuplestorestate)
```

Comment:
other, similar scan nodes in PG also have
```c
    if (node->ss.ps.ps_ResultTupleSlot)
        ExecClearTuple(node->ss.ps.ps_ResultTupleSlot);

    ExecScanReScan(&node->ss);
```
do we need that there too?

Reply:
Yep. We need to free slot for the heap tuple. And `ExecScanReScan` must be called according to the comments on the function. (related to `DELETE/UPDATE <tbl> WHERE CURRENT OF <cursor>` syntax, which we do not support yet)